### DISCUSSION TITLE
Flink - A data provider offers Chainlink Oracle service for Filecoin Network

### DISCUSSION THREAD

**Discussion by flyworker** [filecoin-project/community#480] [Project Showcase]
Flink is a data provider DAO aiming at provider Chainlink Oracle service for Filecoin Network. The implementation offered deal information on multi chains for users who want to store there data on Filecoin network.

Flink requires two components to work as a data provider:

* Data aggregator
* Chainlink external adapter
* Data DAO notary

Several blockchain scanner aggeragate the data to a unified data provider, the chainlink esternal adapter with the oracle smartcontracs broadcasting the proof to target blockchain networks. Data notaries will sign the transaction based the oracle data onchain informtion.The payment is triggered once the proof that it has been stored on the Filecoin network is relayed from Chainlink oracles.

https://calibration-mcs.filswan.com

**Comment by flyworker** [comment]
https://github.com/filecoin-project/devgrants/issues/463
Working in progress now.

**Comment by flyworker** [comment]
Code is ready for testing 

https://github.com/filswan/js-mcs-sdk

sdk documentation:

https://docs.filswan.com/multi-chain-storage/app-developer/sdk

**Comment by Stefaan-V** [comment]
That’s awesome work Charles!

Thanks,
Stefaan
